napari/components/_viewer/view/main.py

```diff
@@ -172,6 +172,11 @@ def on_key_release(self, event):
         if layer is not None:
             layer.on_key_release(event)
 
+    def on_draw(self, event):
+        """Called whenever drawn in canvas. Called for all layers, not just top
+        """
+        for layer in self.viewer.layers:
+            layer.on_draw(event)
```

Comment:
How often is this called? It seems like it would be running hundreds of times a second?

Reply:
on draw is called whenever there is a cursor movement, or the scene graph changes. It's not so frequent. i.e. if nothing is happening it doesn't get called, and if something happens it only gets called for each something

Reply:
you mean camera, not cursor movement, right? on cursor seems a bit extreme...

Reply:
yes, camera - sorry!